src/dsql/parse.y

```diff
@@ -7985,13 +8010,110 @@ system_function_special_syntax
 		}
 	| POSITION '(' value_list_opt  ')'
 		{ $$ = newNode<SysFuncCallNode>(*$1, $3); }
+	| rsa_encrypt_decrypt '(' value KEY value opt_lparam opt_hash ')'
+		{
+			$$ = newNode<SysFuncCallNode>(*$1,
+				newNode<ValueListNode>($3)->add($5)->add($6)->
+					add(MAKE_str_constant(newIntlString($7->c_str()), CS_ASCII)));
+			$$->dsqlSpecialSyntax = true;
+		}
+	| RSA_SIGN '(' value KEY value opt_hash opt_saltlen ')'
+		{
+			$$ = newNode<SysFuncCallNode>(*$1,
+				newNode<ValueListNode>($3)->add($5)->
+					add(MAKE_str_constant(newIntlString($6->c_str()), CS_ASCII))->add($7));
+			$$->dsqlSpecialSyntax = true;
+		}
+	| RSA_VERIFY'(' value SIGNATURE value KEY value opt_hash opt_saltlen ')'
+		{
+			$$ = newNode<SysFuncCallNode>(*$1,
+				newNode<ValueListNode>($3)->add($5)->add($7)->
+					add(MAKE_str_constant(newIntlString($8->c_str()), CS_ASCII))->add($9));
+			$$->dsqlSpecialSyntax = true;
+		}
 	| RDB_SYSTEM_PRIVILEGE '(' valid_symbol_name ')'
 		{
 			ValueExprNode* v = MAKE_system_privilege($3->c_str());
 			$$ = newNode<SysFuncCallNode>(*$1, newNode<ValueListNode>(v));
 		}
 	;
 
+%type <metaNamePtr> rsa_encrypt_decrypt
+rsa_encrypt_decrypt
+	: RSA_DECRYPT | RSA_ENCRYPT
+	;
+
+%type <valueExprNode> opt_lparam
+opt_lparam
+	: LPARAM value
+		{ $$ = $2; }
+	| /* nothing */
+		{ $$ = MAKE_str_constant(newIntlString(""), CS_ASCII); }
+	;
+
+%type <metaNamePtr> opt_hash
+opt_hash
+	: HASH valid_symbol_name
+		{ $$ = $2; }
+	| /* nothing */
+		{ $$ = newNode<MetaName>(""); }
+	;
+
+%type <valueExprNode> opt_saltlen
+opt_saltlen
+	: SALT_LENGTH value
+		{ $$ = $2; }
+	| /* nothing */
+		{ $$ = MAKE_str_constant(newIntlString(""), CS_ASCII); }
+	;
+
+%type <metaNamePtr> opt_mode
+opt_mode
+	: MODE valid_symbol_name
+		{ $$ = $2; }
+	| /* nothing */
+		{ $$ = newNode<MetaName>(""); }
+	;
+
+%type <valueExprNode> opt_iv
+opt_iv
```

Comment:
@AlexPeshkoff I prefer to see that `opt_*` things prefixed with something like `crypt_`